### ISSUE TITLE
How Should I Store Image In Jquery Form Builder Because in user data we can't store file,so how i store image,file and show them 

### ISSUE THREAD

**Issue by vijay-torani** [kevinchappell/formBuilder#1476]
### Description:
How to store file 
### Environment Details:

 - formBuilder Version: latest
 - Browser: Chrome
 - OS: Windows

### Expected Behavior
mehtod should be given to store file (image)
### Actual Behavior
mehtod not  be given to store file (image)

**Comment by lucasnetau** [comment]
Take a look at this plugin control

https://github.com/lucasnetau/formBuilder-plugin-media

**Comment by vijay-torani** [comment]
yes i am using this plugin but how to display image when we render form data using userdata method

**Comment by lucasnetau** [comment]
That will require a custom control

**Comment by vijay-torani** [comment]
can you guide me i am new to this plugin  how to add

**Comment by vijay-torani** [comment]
> That will require a custom control

can you guide me i am new to this plugin how to add

**Comment by lucasnetau** [comment]
Please take a look at the documentation for custom controls. There is also my custom controls example repo https://github.com/lucasnetau/formBuilder-plugins which may guide you.

Back of envelope thoughts would be you need to setup both the input[type=file] and likely a hidden input to track the files, and a div to render the images in build() and then in onRender you need to setup your control. You'll need a handler to upload the files to your server and then store a returned filename, then setup the img elements from userData

**Comment by vijay-torani** [comment]
> Please take a look at the documentation for custom controls. There is also my custom controls example repo https://github.com/lucasnetau/formBuilder-plugins which may guide you.
> 
> Back of envelope thoughts would be you need to setup both the input[type=file] and likely a hidden input to track the files, and a div to render the images in build() and then in onRender you need to setup your control. You'll need a handler to upload the files to your server and then store a returned filename, then setup the img elements from userData
can you give any example so i can implement from that because for me is going bit complicated already

**Comment by lucasnetau** [comment]
This is very similar to the request in https://github.com/lucasnetau/formBuilder-plugin-media/issues/1

If you need this, and cannot work it out from the examples I've already provided,  I'm open to sponsorship to build the custom control, it would then be available to the community.

**Comment by vijay-torani** [comment]
thanks but one problem when i am doing dynamic modification to src using javascript on image delete still i am getting src value in formdata 
![Uploading Screenshot 2023-11-03 162536.png…]()

**Comment by vijay-torani** [comment]
![Screenshot 2023-11-03 162554](https://github.com/kevinchappell/formBuilder/assets/107983453/f3d99116-0f16-45b4-9657-ff067098b855)

**Comment by lucasnetau** [comment]
Please don't paste images of code, console etc. use the markdown features of GitHub to highlight the text. It is very hard to read and process code from images. 

Ok, your issue. From memory you cannot dynamically set the src attribute of a input[type=file] for security reasons to prevent rogue script uploading files from a known path.

**Comment by vijay-torani** [comment]
thanks i will now upload like that
